packages/builder/src/builderStore/store/frontend.js

```diff
@@ -85,14 +85,16 @@ const INITIAL_FRONTEND_STATE = {
   selectedScreenId: null,
   selectedComponentId: null,
   selectedLayoutId: null,
-  hoverComponentId: null,
 
   // Client state
   selectedComponentInstance: null,
 
   // Onboarding
   onboarding: false,
   tourNodes: null,
+
+  // UI state
+  hoveredComponentId: null,
```

Comment:
We could possibly go one step further with this and mirror the dedicated hover store approach you made in the client. The constant updates to the builderStore also triggers quite a bit of computation on the builder side. This could also be addressed in the monolith ticket.